src/core/core/library_utils.cpp

```diff
@@ -62,7 +62,7 @@ namespace dsn {
             else
                 return true;
 # elif defined(__linux__) || defined(__FreeBSD__) || defined(__APPLE__)
-            module_name += ".dll";
+            module_name += ".so";
```

Comment:
we may use module_name = "lib" + module_name + ".so" instead to ensure in both windows and Linux, the module_name can be given the same. I'm not sure the other platforms such as FreeBSD and Mac though.